.github/workflows/drivers.yml

```diff
@@ -12,8 +12,48 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  # This job will determine if the tier 3 driver tests should be skipped.
+  # It will always run first.
+  determine-skip-tier3-drivers:
+    runs-on: ubuntu-latest
+    outputs:
+      skip_tier3: ${{ steps.check_team_and_branch.outputs.skip }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Read team.json and check if tier3 drivers should run
+        id: check_team_and_branch
+        run: |
+          GITHUB_ACTOR_LOWER=$(echo "${{ github.actor }}" | tr '[:upper:]' '[:lower:]')
+          REQUIRED_TEAM_MEMBERS=$(jq -r '.teams[] | select(.name == "Drivers" or .name == "Querying" or .name == "Semantic Layer") | .members[]' .github/team.json | tr '[:upper:]' '[:lower:]')
+          IS_REQUIRED_MEMBER=false
+          for member in ${REQUIRED_TEAM_MEMBERS}; do
+            if [[ "${GITHUB_ACTOR_LOWER}" == "${member}" ]]; then
+              IS_REQUIRED_MEMBER=true
+              break
+            fi
+          done
+          IS_MASTER_OR_RELEASE="${{ github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/release-') }}"
+          echo "GitHub Actor: ${{ github.actor }}"
+          echo "Team Members: ${REQUIRED_TEAM_MEMBERS}"
+          echo "Is Required Team Member: ${IS_REQUIRED_MEMBER}"
+          echo "Is master or release: "${IS_MASTER_OR_RELEASE}"
+          # Logic to determine if tier 3 tests should skip
+          # Skip if not a required team member AND not on master or release branches
+          if [[ "${IS_REQUIRED_MEMBER}" == "true" ]]; then
+            echo "skip=false >> $GITHUB_OUTPUT"
+            echo "User is a required team member. Tier 3 driver tests will run."
+          elif [[ "${IS_MASTER_OR_RELEASE}" == "true" ]]; then
+            echo "skip=false" >> $GITHUB_OUTPUT
+            echo "Merging into master or a release branch. Tier 3 driver tests will run."
+          else
```

Comment:
could we also add an override for a "run-all" label or commit message so that anyone can opt into running these drivers?

Reply:
if we switch to labels, can they just add the team/drivers label?